packages/flutter/lib/src/cupertino/text_field.dart

```diff
@@ -549,8 +549,24 @@ class _CupertinoTextFieldState extends State<CupertinoTextField> with AutomaticK
     _editableTextKey.currentState.showToolbar();
   }
 
-  void _handleDoubleTapDown(TapDownDetails details) {
-    _renderEditable.selectWord(cause: SelectionChangedCause.tap);
+  void _handleDoubleTapUp(TapUpDetails details) {
+    _renderEditable.selectWord(cause: SelectionChangedCause.doubleTap);
+    _editableTextKey.currentState.showToolbar();
+  }
+
+  void _handleDoubleLongTapStart(LongPressStartDetails details) {
+    _renderEditable.selectWord(cause: SelectionChangedCause.longPress);
```

Comment:
This changes the semantics a bit no? I think iOS selects the word as soon as the second tap is down, rather than holding the second tap for a bit. But I didn't double check.

Reply:
The highlight seems to only appear after a short interval, or upon touch up.